plugins/woocommerce/includes/tracks/events/class-wc-products-tracking.php

```diff
@@ -135,22 +135,28 @@ public function track_product_updated_client_side( $post ) {
 			if ( $( 'h1.wp-heading-inline' ).text().trim() === '" . __( 'Edit product', 'woocommerce' ) . "') {
 				var initialStockValue = $( '#_stock' ).val();
 				var hasRecordedEvent = false;
+				var isBlockEditor = false;
 
-				$( '#publish' ).on( 'click', function() {
+				function sendTracks() {
 					if ( hasRecordedEvent ) {
 						return;
 					}
 
+					if ( $( '.block-editor' ).length !== 0 && $( '.block-editor' )[0] ) {
```

Comment:
We do this check for both button selectors to account for any naming variations, e.g. can a theme with block based editing enabled have a "publish" selector for the Update button?.